### PR TITLE
[core] Deprecations in CPDConfiguration, Base(CPD)CLITest, AbstractAntTestHelper

### PR DESCRIPTION
**Pull request by adangel**
## Describe the PR

* Adds the deprecation notes for #4235 
* Also adds some deprecations for #4155

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">Compared to master:<br />
This changeset changes 0 violations,<br />
introduces 0 new violations, 0 new errors and 0 new configuration errors,<br />
removes 0 violations, 0 errors and 0 configuration errors.<br />
<a href="https://chunk.io/pmd/f2224f7f267d4341b7ad10e0f5be4154/diff1/index.html">Full report</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: pmd-core/src/main/java/net/sourceforge/pmd/cpd/CPDConfiguration.java [hunk 1]
```diff
@@ -144,7 +142,11 @@ public class CPDConfiguration extends AbstractConfiguration {
     @Parameter(names = { "--debug", "--verbose", "-v", "-D" }, description = "Debug mode.")
     private boolean debug = false;
 
-    // this has to be a public static class, so that JCommander can use it!
+    /**
+     * @deprecated this is internal API. The class has to be public static, so that JCommander can use it.
+     */
+    @Deprecated
+    @InternalApi
```
Comment (jsotuyod) [inline pmd-core/src/main/java/net/sourceforge/pmd/cpd/CPDConfiguration.java:149]: this doesn't even need to be marked as `@InternalApi`, we are not moving it to an `internal` package in PMD 7, but removing it altogether.

File: pmd-core/src/main/java/net/sourceforge/pmd/cpd/CPDConfiguration.java [hunk 2]
```diff
@@ -156,12 +158,21 @@ public Language convert(String languageString) {
...
-        this.encoding = encoding;
         setSourceEncoding(encoding);
     }
 
+    @Parameter(names = { "--encoding", "-e" }, description = "Character encoding to use when processing files", required = false)
+    @Override
+    public void setSourceEncoding(String sourceEncoding) {
+        super.setSourceEncoding(sourceEncoding);
+    }
+
```
Comment (jsotuyod) [inline pmd-core/src/main/java/net/sourceforge/pmd/cpd/CPDConfiguration.java:175]: what's the point of adding this override? This method should still be removed in PMD 7.

Comment (adangel) [reply pmd-core/src/main/java/net/sourceforge/pmd/cpd/CPDConfiguration.java:175]: This is to add the annotation `@Parameter` to this method...

Comment (jsotuyod) [reply pmd-core/src/main/java/net/sourceforge/pmd/cpd/CPDConfiguration.java:175]: yeah, but why not keep it on the other (deprecated) method? We are going to remove both, the method and the annotation as part of PMD 7, as JCommander is no more.

File: pmd-core/src/main/java/net/sourceforge/pmd/cpd/CPDConfiguration.java (hunk 1 above)
Comment (adangel) [reply pmd-core/src/main/java/net/sourceforge/pmd/cpd/CPDConfiguration.java]: done

File: pmd-core/src/main/java/net/sourceforge/pmd/cpd/CPDConfiguration.java (hunk 2 above)
Comment (adangel) [reply pmd-core/src/main/java/net/sourceforge/pmd/cpd/CPDConfiguration.java]: done

**Pull request merged**